ui/app/AppLayouts/Chat/ChatColumn.qml

```diff
@@ -63,6 +63,57 @@ StackLayout {
             }
         }
  
+        ListModel {
+            id: suggestions
+        }
+
+        Connections {
+            target: chatsModel
+            onActiveChannelChanged: {
+              suggestions.clear()
+              for (let i = 0; i < chatsModel.suggestionList.rowCount(); i++) {
+                suggestions.append({
+                  alias: chatsModel.suggestionList.rowData(i, "alias"),
+                  ensName: chatsModel.suggestionList.rowData(i, "ensName"),
+                  address: chatsModel.suggestionList.rowData(i, "address"),
+                  identicon: chatsModel.suggestionList.rowData(i, "identicon"),
+                  ensVerified: chatsModel.suggestionList.rowData(i, "ensVerified")
+                });
+              }
+            }
+        }
+
+        SuggestionBox {
+            id: suggestionsBox
+            model: suggestions
+            width: chatContainer.width
+            anchors.bottom: inputArea.top
+            anchors.left: inputArea.left
+            filter: chatInput.textInput.text
+            property: "ensName, alias"
+            onItemSelected: function (item) {
+              let currentText = chatInput.textInput.text
+              let lastAt = currentText.lastIndexOf("@")
+              let aliasName = item[suggestionsBox.property.split(",").map(p => p.trim()).find(p => !!item[p])]
+              let nameLen = aliasName.length + 2 // We're doing a +2 here because of the `@` and the trailing whitespace
+              let position = 0;
+              let text = ""
+
+              if (currentText.length == 1) {
+                position = nameLen
+                text = "@" + aliasName + " "
+              } else {
+                let left = currentText.slice(0, lastAt)
+                position = left.length + nameLen
+                text = left + "@" + aliasName + " "
+              }
+
+              chatInput.textInput.text = text
+              chatInput.textInput.cursorPosition = position
+              suggestionsBox.suggestionsModel.clear()
+            }
+        }
```

Comment:
BTW, I can tell you wrote that code outside of QT Creator because you used 2 spaces. For QML, we use 4